### DISCUSSION TITLE
[Q] Separate worker for a separate queue

### DISCUSSION THREAD

**Discussion by matvey417** [roadrunner-server/roadrunner#889] [Q&A]
Sorry for my english.
I set up RR, selected workers, added queues, everything is super!
But for a specific task, I want to have its own queue, and this queue always has its own worker.
Separate worker for a separate queue.
How can this be done as simply as possible?

**Comment by rustatian** [comment]
Hey @matvey417. You can use the `kv` plugin with `memory` driver to bind specific payloads to a specific workflow. For example, you may store the `queue` name in the in-memory [KV](https://roadrunner.dev/docs/beep-beep-kv) and when the request arrived, you may identify this request with KV stored data and redirect the particular payload to the needed subsystem.
Or you may use the `temporal` plugin https://docs.temporal.io/docs/php/introduction/ to bind specific payloads to a specific workflow. Here you can find a lot of samples: https://github.com/temporalio/samples-php

**Reply by matvey417** [reply]
Great, thank you.
Is there no easier way to solve this problem?

these are too cumbersome solutions for my initial task.
I would not like to install large plugins to solve a small task

**Reply by rustatian** [reply]
You may use the `KV` with `memory` driver. It has 0 dependencies and you only need to specify it in the configuration.

```yaml
kv:
  memory-rr:
    driver: memory
    config:
      interval: 1
```

If you want a different solution, you may ask the community in our [Discord server](https://discord.gg/TFeEmCs) (RU/EN) or issue a feature request.
